daal4py/sklearn/_utils.py

```diff
@@ -129,6 +129,10 @@ def sklearn_check_version(ver):
     return res
 
 
+if sklearn_check_version("1.2"):
+    from sklearn.utils._param_validation import validate_parameter_constraints
```

Comment:
Better to move it on top of the file.

Reply:
`sklearn_check_version` is defined in same file after all other imports.